cob_script_server/src/simple_script_server/simple_script_server.py

```diff
@@ -862,27 +886,31 @@ def say(self,component_name,parameter_name,blocking=True):
 		# get values from parameter server
 		language = rospy.get_param(self.ns_global_prefix + "/" + component_name + "/language","en")
 		if type(parameter_name) is str:
-			if not rospy.has_param(self.ns_global_prefix + "/" + component_name + "/" + language + "/" + parameter_name):
-				rospy.logerr("parameter %s does not exist on ROS Parameter Server, aborting...",self.ns_global_prefix + "/" + component_name + "/" + language + "/" + parameter_name)
-				ah.set_failed(2)
+			full_parameter_name = self.ns_global_prefix + "/" + component_name + "/" + language + "/" + parameter_name
+			if not rospy.has_param(full_parameter_name):
+				message = "Parameter " + full_parameter_name + " does not exist on ROS Parameter Server, aborting..."
+				rospy.logerr(message)
+				ah.set_failed(2, message)
 				return ah
-			param = rospy.get_param(self.ns_global_prefix + "/" + component_name + "/" + language + "/" + parameter_name)
+			param = rospy.get_param(full_parameter_name)
 		else:
 			param = parameter_name
 
 		# check parameters
 		if not type(param) is list: # check list
-				rospy.logerr("no valid parameter for %s: not a list, aborting...",component_name)
-				print "parameter is:",param
-				ah.set_failed(3)
-				return ah
+			message = "No valid parameter for " + component_name + ": not a list, aborting..."
+			rospy.logerr(message)
+			print "parameter is:",param
```

Comment:
again a `print`... there are some more in here as well, which you didn't touch. Change, leave or set up an issue? I don't mind...